src/client.rs

```diff
@@ -191,6 +196,17 @@ pub struct Client {
     sync_if_empty: bool,
     #[builder(default = "true")]
     trust_local: bool,
+    #[builder(private, default = "Cell::new(SyncState::Ok)")]
+    sync_state: Cell<SyncState>,
+}
+
+#[derive(Clone, Copy, Debug)]
+pub enum SyncState {
+    Ok,
+    Backoff {
+        observed_at: Instant,
+        duration: Duration,
+    },
```

Comment:
I think we could make do with simply storing `until: Instant`, or even easier, ditch the whole struct and use `backoff_until: Option<Instant>` instead.